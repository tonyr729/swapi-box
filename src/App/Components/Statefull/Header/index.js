import React, { Component } from 'react';
import './index.css'

class Header extends Component {
  constructor() {
    super();
    this.state = {

    }
  }

  render() {
    return (
      <div>
        <h1 className="header-title">SWAPI-Box</h1>
        <button>View Favorites 0</button>
        <button>People</button>
        <button>Planets</button>
        <button>Vehicles</button>
      </div>
    )
  }
}

export default Header;