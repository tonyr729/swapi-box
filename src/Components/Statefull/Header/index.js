import React, { Component } from 'react';
import APIManager from '../../../API/Helpers/API Manager/apiManager'
import './index.css'
const apiManager = new APIManager();


class Header extends Component {
  constructor() {
    super();
    this.state = {

    }
  }

  componentDidMount() {
   
  }

  render() {
    return (
      <div>
        <h1 className="header-title">SWAPI-Box</h1>
        <button onClick>View Favorites 0</button>
        <button onClick={apiManager.fetchPeople()}>People</button>
        <button>Planets</button>
        <button>Vehicles</button>
      </div>
    )
  }
}

export default Header;