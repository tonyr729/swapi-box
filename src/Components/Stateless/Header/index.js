import React, { Component } from 'react';
import './index.css'
import FavButton from '../FavButton/FavButton'


const Header = ({setPeopleData, setPlanetData, setVehicleData, setDisplayedData, favorites}) =>{
 
  return (
    <div>
      <h1 className="header-title">SWAPI-Box</h1>
      <FavButton favorites={ favorites } setDisplayedData={ setDisplayedData } />
      <button onClick={ setPeopleData }>People</button>
      <button onClick={ setPlanetData }>Planets</button>
      <button onClick={ setVehicleData }>Vehicles</button>
    </div>
  )
}


export default Header;