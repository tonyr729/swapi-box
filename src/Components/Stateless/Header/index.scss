
.header {
  display: flex;
  flex-wrap: wrap;
}

.header-title {
  color: #fff;
  margin-left: 1rem;
}

.button-container {
  border: 1px solid orange;
  width: 100%;
}

button {
  font-size: 1rem;
  padding: 0 .6rem;
  height: 3rem;
  float: left;
  border-style: none;
  border-right: 1px solid rgb(255, 165, 0);
  background-color: transparent;
  color: white;
  &:hover { background-color: rgb(228, 148, 0); }
  &:active { background-color: rgb(255, 81, 1); }
}

