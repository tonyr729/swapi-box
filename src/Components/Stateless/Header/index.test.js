import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './index'

describe('Header', () => {

  describe('Component', () => {
    it('should match snapshot', () => {
      const header = renderer.create(<Header />).toJSON();

      expect(header).toMatchSnapshot();
    })
  })
})