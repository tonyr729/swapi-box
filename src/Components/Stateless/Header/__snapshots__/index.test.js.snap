// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<div>
  <h1
    className="header-title"
  >
    SWAPI-Box
  </h1>
  <div
    onClick={[Function]}
  >
    <p
      className="favorite-text"
    >
      Favorites
    </p>
    <div>
      <p
        className="favorite-count"
      />
    </div>
  </div>
  <button
    onClick={undefined}
  >
    People
  </button>
  <button
    onClick={undefined}
  >
    Planets
  </button>
  <button
    onClick={undefined}
  >
    Vehicles
  </button>
</div>
`;
